routes/donations.js

```diff
@@ -0,0 +1,168 @@
+const express = require('express');
+const { keysToCamel } = require('../common/utils');
+const { pool, db } = require('../server/db');
+
+const router = express.Router();
+
+// get all donation rows
+router.get('/', async (req, res) => {
+  try {
+    const allDonations = await pool.query(`SELECT * FROM donations`);
+    res.status(200).json(keysToCamel(allDonations.rows));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// get specific donation
+router.get('/:donationId', async (req, res) => {
+  try {
+    const { donationId } = req.params;
+    const donation = await pool.query(`SELECT * from donations WHERE id = $1`, [donationId]);
+    res.status(200).json(keysToCamel(donation.rows));
+  } catch (err) {
+    res.status(400).send(err.message);
+  }
+});
+
+// create new donation
+router.post('/', async (req, res) => {
+  try {
+    const {
+      id,
+      routeId,
+      orderNum,
+      status,
+      addressStreet,
+      addressUnit,
+      addressCity,
+      addressZip,
+      firstName,
+      lastName,
+      email,
+      phoneNum,
+      notes,
+      date,
+    } = req.body;
+    const donation = await db.query(
+      `INSERT INTO donations (
+        ${routeId ? 'route_id, ' : ''}
+        ${orderNum ? 'order_num, ' : ''}
+        status, address_street,
+        ${addressUnit ? 'address_unit, ' : ''}
+        address_city, address_zip, first_name,
+        last_name, email, phone_num,
+        ${notes ? 'notes, ' : ''}
+        ${date ? 'date, ' : ''}
+        id
```

Comment:
Because the `id` is `SERIAL`, we don't need to add this as a value; PostgreSQL will automatically add it. We also don't need to be passing it as a variable.